File: src/main/kotlin/cz/uhk/ppro/mhjp/resftulgallery/dto/ImageDto.kt
package cz.uhk.ppro.mhjp.resftulgallery.dto

class NewImageDto(
        val description: String  = "",
        val imageBytes: ByteArray = byteArrayOf(),
        val private: Boolean = false
)

class UpdateImageDto(
        val description: String = "",
        val private: Boolean = false
)

open class NewImageDataDto(
        val uuid: String,
        val description: String,
        val imageBytes: ByteArray,
        val private: Boolean,
        val author: ListUserDataDto
) : DataDto()

open class NewAnonImageDataDto(
        val uuid: String,
        val deleteHash: String,
        val description: String,
        val imageBytes: ByteArray
) : DataDto()

open class ImageDataDto(
        val uuid: String,
        val description: String,
        val imageBytes: ByteArray,
        val private: Boolean,
        val author: ListUserDataDto,
        val likedByUsers: List<DataDto>,
        val comments: List<DataDto>
) : DataDto()

open class AnonImageDataDto(
        val uuid: String,
        val description: String,
        val imageBytes: ByteArray,
        val likedByUsers: List<DataDto>,
        val comments: List<DataDto>
) : DataDto()

open class ListImageDataDto(
        val uuid: String
) : DataDto()

class ImagesListDto(
        val images: List<ListImageDataDto>
) : DataDtoWithoutLinks()

File: src/main/bobril/cz/uhk/ppro/mhjp/gallery/api/dto/imageDto.ts
import {DataDto, DataDtoWithoutLinks} from "./baseDto";
import {ListUserDataDto} from "./UserDto";

export interface NewImageDto{
        description: string,
        imageBytes: string[],
        private: boolean
}

export interface UpdateImageDto{
        description: string ,
        private: boolean
}

export interface NewImageDataDto{
        uuid: string,
        description: string,
        imageBytes: string[],
        private: boolean,
        author: ListUserDataDto
}

export interface NewAnonImageDataDto{
        uuid: string,
        deleteHash: string,
        description: string,
        imageBytes: string[]
}

export interface ImageDataDto{
        uuid: string,
        description: string,
        imageBytes: string[],
        private: boolean,
        author: ListUserDataDto,
        likedByUsers: DataDto[],
        comments: DataDto[]
}

export interface AnonImageDataDto{
        uuid: string,
        description: string,
        imageBytes: string[],
        likedByUsers: DataDto[],
        comments: DataDto[]
}

export interface ListImageDataDto{
        uuid: string
}

export interface ImagesListDto extends DataDtoWithoutLinks{
        images: ListImageDataDto[]
}